src/base_event.cpp

```diff
@@ -58,11 +58,22 @@ void MainFrame::OnLoadFiles(wxCommandEvent& event){
 }
 
 void MainFrame::enableGuiElements(bool inp){
-  // deactivate the Start and Load files buttons during the calculation
+  // deactivate the calculation and reload buttons during the calculation
   calcButton->Enable(inp);
   loadFilesButton->Enable(inp);
   // prevents editing of the atom list grid during the calculation
   atomListGrid->EnableEditing(inp);
+  // keeps calculation button deactivated if no valid structure is loaded
+  if (atomListGrid->GetNumberRows() == 0) {
+    calcButton->Enable(false);
+  }
+  // activates pdb specific options only if pdb file is loaded
+    if (getAtomFilepath().size() > 3 && getAtomFilepath().substr(getAtomFilepath().size()-4, 4) == ".pdb"){
+      pdbHetatmCheckbox->Enable(true);
+    }
+    else {
+      pdbHetatmCheckbox->Enable(false);
+    }

```

Comment:
I thought like you initially but the purpose of disabling gui elements during algorithms runs is to avoid the user to mess up with any parameter.

For example, imagine that the user starts a calculation then, during the process, he unticks the "analyze crystal unit cell" option. We should probably extend the deactivation to other parameters such as the grid step and octree level.
Before you argue that the "include HETATM" checkbox has no effect on the calculation button, I would like to point out that I think about future features in advance and that the calculation button will ultimately generate the report file which will include the options/parameters used such as `structure file: xxx` and `HETATM lines included`.
In the end, we should make a more general function disabling all modifiable gui elements when any of the back-end algorithms is running.

Regarding the calculation button. The issue is that the previous disabling/enabling function enabled the "calculation" button after trying to load files even if no valid structure was loaded which is abnormal. So in case no atom is found in the structure file, the calculation button now remains disabled.

Reply:
We can leave it as it is for now, but I've opened an issue about this #39.